compiler/rustc_typeck/src/expr_use_visitor.rs

```diff
@@ -590,11 +625,10 @@ impl<'a, 'tcx> ExprUseVisitor<'a, 'tcx> {
 fn copy_or_move<'a, 'tcx>(
     mc: &mc::MemCategorizationContext<'a, 'tcx>,
     place_with_id: &PlaceWithHirId<'tcx>,
+    diag_expr_id: hir::HirId,
 ) -> ConsumeMode {
-    if !mc.type_is_copy_modulo_regions(
-        place_with_id.place.ty(),
-        mc.tcx().hir().span(place_with_id.hir_id),
-    ) {
+    if !mc.type_is_copy_modulo_regions(place_with_id.place.ty(), mc.tcx().hir().span(diag_expr_id))
```

Comment:
Honestly not sure if we should be using `diag_expr_id` or `place_with_id.hir_id`